test/server/core/card/CardHighlighting.spec.ts

```diff
@@ -0,0 +1,67 @@
+describe('Card highlighting', function() {
+    integration(function(contextRef) {
+        describe('Card highlighting', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        resources: ['armed-to-the-teeth',
+                            'collections-starhopper',
+                            'covert-strength',
+                            'chewbacca#pykesbane',
+                            'battlefield-marine',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                        ],
+                        hand: ['vanguard-infantry'],
+                        groundArena: ['wampa'],
+                        spaceArena: ['cartel-spacer'],
+                    },
+                    player2: {
+                        groundArena: ['death-trooper'],
+                        spaceArena: ['mercenary-gunship']
+                    }
+                });
+            });
+
+            it('the prompt before an action and after should be different.', function () {
+                const { context } = contextRef;
+                let currentPossibleActions = context.player1.currentPrompt();
+
+                const reset = () => {
+                    currentPossibleActions = context.player1.currentPrompt();
+                };
+
+                // attack action
+                context.player1.clickCard(context.wampa);
+                context.player1.clickCard(context.deathTrooper);
+                context.player2.claimInitiative();
+                expect(context.player1.currentPrompt()).not.toEqual(currentPossibleActions);
```

Comment:
We should specifically test that the expected action was removed, could use `.filter()` to select items that are in one array but not the other

Reply:
alrighty!

Reply:
do you have an example of this perhaps?

Reply:
Assuming we have arrays for before and after the action, I think the easiest way to do this check without implementing full array difference would be:

```typescript
expect(beforeActions.length - afterActions.length).toBe(1);
expect(beforeActions.filter((beforeAction) => !afterActions.includes(beforeAction));
```

I'm realizing that's a little slow though since it's M * N complexity. so probably easiest just to determine the name of the action and explicitly check that it's not in the list anymore

Reply:
gotcha!